### ISSUE TITLE
Plugin fails to create injector after upgrade to Xtext 2.8.4v1001508050135

### ISSUE THREAD

**Issue by jimidle** [jknack/antlr4ide#131]
Eclipse Luna just upgraded my XText installation to Xtext 2.8.4v1001508050135 and in typical fashion for Xtext as they seem to change major interfaces in very minor version increments, this means that the ANTLR IDE no longer works. It needs rebuilding with the latest XText i suspect, so I will give it a go, but it needs upgrading for the Marketplace installation of course.

Trivial to reproduce - just try to open a parser grammar and you will get:

```
Failed to create injector for com.github.jknack.antlr4ide.Antlr4 ExtensionFactory: com.github.jknack.antlr4ide.ui.Antlr4ExecutableExtensionFactory

org.eclipse.core.runtime.CoreException: Failed to create injector for com.github.jknack.antlr4ide.Antlr4 ExtensionFactory: com.github.jknack.antlr4ide.ui.Antlr4ExecutableExtensionFactory
at     org.eclipse.xtext.ui.guice.AbstractGuiceAwareExecutableExtensionFactory.create(AbstractGuiceAwareExecutableExtensionFactory.java:59)
at org.eclipse.core.internal.registry.ConfigurationElement.createExecutableExtension(ConfigurationElement.java:262)
at org.eclipse.core.internal.registry.ConfigurationElementHandle.createExecutableExtension(ConfigurationElementHandle.java:55)
at org.eclipse.ui.internal.WorkbenchPlugin$1.run(WorkbenchPlugin.java:293)
at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
at org.eclipse.ui.internal.WorkbenchPlugin.createExtension(WorkbenchPlugin.java:288)
    etc...
```

Unfortunately, that makes it unusable of course. I hope you are still working on this - I see quite a number of issues have been opened without any comments.

**Comment by icar-aose** [comment]
same for me... do you know how to downgrade Xtext to a working version?

**Comment by RobinDeWalvis** [comment]
you can get an older version of XText here: https://www.eclipse.org/modeling/tmf/downloads/index.php?project=xtext&showAll=1&showMax=5&sortBy=
which you can install in the "install new software" section.

**Comment by jeromefroe** [comment]
I encountered the same error with Xtext 2.9.0. Could you provide more details on how you downgraded your version of Xtext? When I enter the above URL I get an error message saying that Eclipse cannot find any software at the given site.

**Comment by jimidle** [comment]
Every xtext upgrade, however minor, seems to break every plugin I have
built with it - I don't blame the author. I think it needs a rebuild with
Xtext 2.9.0.

**Comment by rensink** [comment]
I didn't have to actually downgrade XText, installing version 2.7.3 it over the plugin did the trick. The link given by @Snektron works just fine, note that you have to download the zip file and then use "Install new Software -> Add -> Archive"

I guess first installing XText and then ANTLR would be even better, as then the ANTLR plugin won't feel the need to also get the (newest, and therefore wrong version of) XText.

**Comment by arifogel** [comment]
I've patched this issue in Luna, at least. I have yet to test in Mars, since Mars sucks on my machine.. See my pull request: #144. You'll have to use XText 2.9.2

**Comment by jimidle** [comment]
I can test it in Mars.2, but I will let it come through the pipes.

XText sucks really. It breaks half the plugins that use it on every small
point upgrade. They have no idea about API backwards compatibility etc and
their decision to 'improve' on the ANTLR grammar makes it a horrible task
to maintain both a compiler and a plugin for it. Ah well. At least this
plugin is a good one :)

**Comment by arifogel** [comment]
Tested in Mars. I've created a working release. See #136

**Comment by arifogel** [comment]
Fixed by pull request #144 .
